tests/unit/test_timeparser.py

```diff
@@ -0,0 +1,122 @@
+import pytest
+
+import timeparser
+import datetime
+
+time_delta_test_params = [(datetime.timedelta(days=1), 86400),
+                          (datetime.timedelta(
+                              days=1, hours=3, minutes=15, seconds=32), 98132),
+                          (datetime.timedelta(hours=1, minutes=10), 4200),
+                          (datetime.timedelta(hours=-1), -3600), (None, 0)]
+
+
+@pytest.mark.parametrize('delta,expect', time_delta_test_params)
+def test_time_delta_2_second(delta, expect):
+    ''' Test timeDelta2secs function, convert time delta object to seconds
+    Normal cases:
+    case 1: time delta is 1 day, expect is 86400
+    case 2: time delta is 1 day 3 hour 15 minutes 32 seconds, expect is 98132
+    case 3: time delta is less than 1 day, only 1 hour 10 minutes, expect is 4200
+    case 4: time delta is 1 hour ago, expect is -3600
+
+    Corner cases:
+    case 1: delta object is None -- invalid input, expect is <TBD>
```

Comment:
@arctan5x @lephino @jmeixensperger 
I put the case description in the doc string. 
For the time parser help functions, there are some corner cases about empty input argument value, such as empty string or None. Please review if these cases are valid or not. 
Probably for the internal used functions, the None input case is not necessary?

Let me know your opinions about them. Thanks.

Reply:
I agree with you @GordonWang, some of the internal functions will never use those types of inputs. I'd suggest making these corner cases for inputs that can be changed by the user's configuration. Out of the cases in this PR marked with <TBD>, I think that the only one we really need is this "case 1" directly above.
For the output, I'm not as sure what we should expect for this case. 0 would be okay with me, as long as we are logging an error message for clarity.